packages/instantsearch-react/src/utils.js

```diff
@@ -12,3 +12,20 @@ export function isSpecialClick(event) {
 export function capitalize(key) {
   return key.length === 0 ? '' : `${key[0].toUpperCase()}${key.slice(1)}`;
 }
+
+export function assertFacetDefined(searchParameters, searchResults, facet) {
+  const wasRequested =
+    searchParameters.isConjunctiveFacet(facet) ||
+    searchParameters.isDisjunctiveFacet(facet);
```

Comment:
I'm not using this function in connectHierarchicalMenu yet, so I haven't updated it for that purpose. I think if we wanted to check that a hierarchical facet is correctly defined on the server side, we'd need to test every of its `attributes` properties on the response?